# import necessary libraries

In [2]:
import pandas as pd
import numpy as np
from scipy import stats

##  Business Problem
Is there significant difference in the diameter of the cutlet ?


## Data description

α is given as 0.05 (95% Confidence)

Y is a Continious Data 
X is a Discrete Data

Are Y1 and Y2 normal ?

H0 = Y1 and Y2 are normal  
H1 = Y1 and Y2 are not normal


## __3 - Normality Test__  

In [3]:
df = pd.read_csv("Cutlets.csv")

In [6]:
shapiro_unit_a = stats.shapiro(df["Unit A"])
shapiro_unit_a

ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)

P value for Unit A == 0.32 > α

In [7]:
shapiro_unit_b = stats.shapiro(df["Unit B"])
shapiro_unit_b

ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)

(0.9727300405502319, 0.5224985480308533)  
P value for Unit B == 0.52 > α

### HO is accepted. Thats is both Y1 and Y2 are normal

## __4 - External Condition__

### External condition is same.  
### By the above analysis we can perform Paired T Test

H0 == Mean for Y1 and Y2 are equal (There is no significance difference between diameter of the Culets)  
H1 == Mean for Y1 and Y2 are not equal (There is a significance difference between diameter of the Culets)

In [8]:
stats.ttest_rel(df["Unit A"], df["Unit B"])

Ttest_relResult(statistic=0.7536787225614314, pvalue=0.4562300768038412)

Ttest_relResult(statistic=0.7536787225614323, pvalue=0.45623007680384076)  
P value of the Paired T Test is == 0.45 > α  

## Thus H0 is accepted.  
## Mean of both Y1 and Y2 are equal